pkg/cmd/enable/addons/exec.go

```diff
@@ -0,0 +1,152 @@
+// Copyright Contributors to the Open Cluster Management project
+package addons
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	clusterclientset "open-cluster-management.io/api/client/cluster/clientset/versioned"
+	"open-cluster-management.io/clusteradm/pkg/cmd/enable/addons/scenario"
+	"open-cluster-management.io/clusteradm/pkg/helpers"
+	"open-cluster-management.io/clusteradm/pkg/helpers/apply"
+)
+
+//ClusterName: The cluster name used in the template
+type ClusterName struct {
+	ClusterName string
+}
+
+func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
+	klog.V(1).InfoS("addon options:", "dry-run", o.ClusteradmFlags.DryRun, "names", o.names, "clusters", o.clusters, "output-file", o.outputFile)
+
+	return nil
+}
+
+func (o *Options) validate() error {
+	if o.names == "" {
+		return fmt.Errorf("names is missing")
+	}
+
+	names := strings.Split(o.names, ",")
+	for _, n := range names {
+		if n != "application-manager" {
+			return fmt.Errorf("invalid add-on name %s", n)
+		}
+	}
+
+	if o.clusters == "" {
+		return fmt.Errorf("clusters is misisng")
+	}
+
+	return nil
+}
+
+func (o *Options) run() error {
```

Comment:
should we have some integration test around this?

Reply:
done as suggested.